parla/tasks.py

```diff
@@ -447,6 +447,124 @@ def get_current_devices() -> List[Device]:
     return list(task_runtime.get_devices())
 
 
+@contextmanager
+def _reserve_persistent(memsize, device):
+    resource_pool = get_scheduler_context().scheduler._available_resources
+    resource_pool.allocate_resources(device, {'memory' : memsize}, blocking = True)
+    try:
+        yield
+    finally:
+        resource_pool.deallocate_resources(device, {'memory' : memsize})
+
+
+def _get_parla_device(device):
+    if isinstance(device, Device):
+        return device
+    try:
+        import cupy
+    except ImportError:
+        pass
+    else:
+        if isinstance(device, cupy.cuda.Device):
+            from .cuda import gpu
+            index = device.id
+            return gpu(index)
+    raise ValueError("Don't know how to convert object of type {} to a parla device object.".format(type(device)))
+
+
+@contextmanager
+def reserve_persistent_memory(amount, device = None):
+    """
+    :param amount: The number of bytes reserved in the scheduler from tasks for persitent data. \
+      This exists, not as any kind of enforced limit on allocation, but rather to let the scheduler \
+      have an accurate measure of memory occupancy on the GPU beyond just memory that's used \
+      only during a task's execution. It can be specified as an integer representing the nubmer of \
+      bytes, an ndarray (cupy or numpy), or a list of ndarrays.
+    :param device: The device object where memory is to be reserved. \
+      This must be supplied if amount is an integer \
+      and may be supplied for an array. In the case of a list or other iterable it must \
+      be supplied if any element of the list is not an array. This may be a list of \
+      devices if amount is a list of array objects.
+    """
+    from . import cpu
+    if isinstance(amount, int):
+        memsize = amount
+    elif hasattr(amount, '__cuda_array_interface__'):
+        import cupy
+        # cupy has the info we need and views are cheap,
+        # so get a view then read off the size/device info.
+        view = cupy.array(amount)
+        memsize = view.nbytes
+        if device is None:
+            device = view.device
```

Comment:
I think you can just use `storage_size` here. It's in `parla.array` I think.

Reply:
Good point. I wasn't aware of that function.